### ISSUE TITLE
Mustache templates are holding us back!

### ISSUE THREAD

**Issue by brndnmtthws** [mesosphere/universe#269]
Mustache templates are very limited (by design), and IMO make it difficult to implement certain features in the Universe. I'd like it if we could explore some more useful templating systems. For example, dealing with trailing commas in JSON (such as in a list or set, or at the end of a JSON block) with mustache is pretty much impossible.

**Comment by jsancio** [comment]
Yes @brndnmtthws, this has come up a couple of times. Either have a more expressive template engine or a domain specific language.

**Comment by cmaloney** [comment]
For DCOS Image we recently added a new template language which is a little more expressive and designed so that the templates can be the source of truth for what configuration variables need to be set (Almost all current template systems work on some variant of MVC, where the template is just something you use to make a view of your actual data store), might be worthwhile to see if it would solve your same use cases

**Comment by brndnmtthws** [comment]
@cmaloney is that a third party library? Or did we write our own?

**Comment by cmaloney** [comment]
Wrote our own after evaluating a lot of different ones.

**Comment by BenWhitehead** [comment]
follow
